### ISSUE TITLE
Modify validator assignment algorithm to support state sync

### ISSUE THREAD

**Issue by mzhangmzz** [near/nearcore#8505]
We would like to allow chunk only producers to only track one shard, which requires the implementation of state sync. This issues tracks the protocol changes needed to enable state sync. 

- [ ] Make the chunk assignment algorithm stickier
- [ ] Validators should be able to choose whether to be block producers or chunk producers. And the reward algorithm should reward block producers more for the additional work they do.

**Comment by walnut-the-cat** [comment]
What's the current status of the work? Can we link relevant docs?

**Comment by Longarithm** [comment]
> Make the chunk assignment algorithm stickier

By this we mean that if validator is a chunk producer for shard 0, it stays on this shard for more than one epoch - say, 10 epochs - to trigger state syncs more rarely. But it is lower priority, we need to try speed up state sync first. And it is more natural to increase epoch length if needed.
I don't see obvious implementation caveats. Probably smth local to `EpochManager` especially it knows about all epochs and thus can easily track for how long everyone was a chunk producer. But I never looked into it. Could take <1 month if we understand duration in epochs already?

> Validators should be able to choose whether to be block producers or chunk producers.

Low priority feature, just nice-to-have for ones who cares about specific role, we need to set proper rewards beforehand.

> And the reward algorithm should reward block producers more for the additional work they do.

What was changed after we implemented https://gov.near.org/t/block-and-chunk-producer-selection-algorithm-in-simple-nightshade/66 -> https://www.overleaf.com/project/602e8699bbdc69296c883f36?
Anyway, currently state sync is "infinitely" long so it would need to be rewarded in any model, and we need to estimate state sync work after we improve it.
